classes/source/source_database.php

```diff
@@ -59,8 +59,17 @@ public function extract() {
         global $DB;
 
         $arraytoprocess = $DB->get_records_sql($this->get_settings()['querysql'], null, 0 , $this->get_settings()['querylimit']);
-        $this->data = new data(null, null, $arraytoprocess, null);
 
+        if ($arraytoprocess) {
```

Comment:
Will you be able to an extra setting for headers? Like "First row is a header" or something ...

Reply:
Done